CMakeLists.txt

```diff
@@ -273,9 +273,9 @@ endif ()
 ################################
 ####  Find Python
 ################################
-find_package(PythonInterp 3.5)
-if (PYTHON_VERSION_STRING VERSION_LESS 3.5.0)
-    message(FATAL_ERROR "Python 3.5 or higher is required; found v${PYTHON_VERSION_STRING} instead. Specify interpreter with -DPYTHON_EXECUTABLE=/path/to/python3")
+find_package(Python3 3.5)
```

Comment:
We could probably be much more aggressive with this version number now. These changes need to work with Ubuntu 20.04 but nothing earlier, so think this should be 3.8.

Not relevant for this PR specifically, but @ptheywood what do you think? Could you open a new issue for us to decide an appropriate minimum Python version?

Reply:
<= 3.7 is EOL since summer 2023, so it seems like the sensible current minimum version. 

I've opened #246